File: Heat/Controllers/PlantTypesController.vb
Imports System
Imports System.Collections.Generic
Imports System.Data
Imports System.Data.Entity
Imports System.Linq
Imports System.Net
Imports System.Web
Imports System.Web.Mvc
Imports Heat

Namespace Controllers
    <Authorize> _
    Public Class PlantTypesController
        Inherits System.Web.Mvc.Controller

        Private db As New HeatDBContext

        ' GET: PlantTypes
        Function Index() As ActionResult
            Return View(db.PlantTypes.ToList())
        End Function

        ' GET: PlantTypes/Details/5
        Function Details(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantType As PlantType = db.PlantTypes.Find(id)
            If IsNothing(plantType) Then
                Return HttpNotFound()
            End If
            Return View(plantType)
        End Function

        ' GET: PlantTypes/Create
        Function Create() As ActionResult
            Return View()
        End Function

        ' POST: PlantTypes/Create
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Create(<Bind(Include:="ID,Name")> ByVal plantType As PlantType) As ActionResult
            If ModelState.IsValid Then
                db.PlantTypes.Add(plantType)
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(plantType)
        End Function

        ' GET: PlantTypes/Edit/5
        Function Edit(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantType As PlantType = db.PlantTypes.Find(id)
            If IsNothing(plantType) Then
                Return HttpNotFound()
            End If
            Return View(plantType)
        End Function

        ' POST: PlantTypes/Edit/5
        'To protect from overposting attacks, please enable the specific properties you want to bind to, for 
        'more details see http://go.microsoft.com/fwlink/?LinkId=317598.
        <HttpPost()>
        <ValidateAntiForgeryToken()>
        Function Edit(<Bind(Include:="ID,Name")> ByVal plantType As PlantType) As ActionResult
            If ModelState.IsValid Then
                db.Entry(plantType).State = EntityState.Modified
                db.SaveChanges()
                Return RedirectToAction("Index")
            End If
            Return View(plantType)
        End Function

        ' GET: PlantTypes/Delete/5
        Function Delete(ByVal id As Integer?) As ActionResult
            If IsNothing(id) Then
                Return New HttpStatusCodeResult(HttpStatusCode.BadRequest)
            End If
            Dim plantType As PlantType = db.PlantTypes.Find(id)
            If IsNothing(plantType) Then
                Return HttpNotFound()
            End If
            Return View(plantType)
        End Function

        ' POST: PlantTypes/Delete/5
        <HttpPost()>
        <ActionName("Delete")>
        <ValidateAntiForgeryToken()>
        Function DeleteConfirmed(ByVal id As Integer) As ActionResult
            Dim plantType As PlantType = db.PlantTypes.Find(id)
            db.PlantTypes.Remove(plantType)
            db.SaveChanges()
            Return RedirectToAction("Index")
        End Function

        Protected Overrides Sub Dispose(ByVal disposing As Boolean)
            If (disposing) Then
                db.Dispose()
            End If
            MyBase.Dispose(disposing)
        End Sub
    End Class
End Namespace

File: Heat.ConvertedToC#/Controllers/PlantTypesController.cs
using System.Linq;
using System.Web.Mvc;
using System.Data.Entity;
using System.Net;
using Heat.Repositories;
using Heat.Models;

namespace Heat.Controllers
{
    [Authorize()]
	public class PlantTypesController : System.Web.Mvc.Controller
	{


		private IHeatDBContext _db ;
        public PlantTypesController(IHeatDBContext context)
        {
            _db = context;
        }

		// GET: PlantTypes
		public ActionResult Index()
		{
			return View(_db.PlantTypes.ToList());
		}

		// GET: PlantTypes/Details/5
		public ActionResult Details(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantType plantType = _db.PlantTypes.Find(id);
			if ((plantType == null)) {
				return HttpNotFound();
			}
			return View(plantType);
		}

		// GET: PlantTypes/Create
		public ActionResult Create()
		{
			return View();
		}

		// POST: PlantTypes/Create
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Create(		[Bind(Include = "ID,Name")]
PlantType plantType)
		{
			if (ModelState.IsValid) {
				_db.PlantTypes.Add(plantType);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(plantType);
		}

		// GET: PlantTypes/Edit/5
		public ActionResult Edit(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantType plantType = _db.PlantTypes.Find(id);
			if ((plantType == null)) {
				return HttpNotFound();
			}
			return View(plantType);
		}

		// POST: PlantTypes/Edit/5
		//To protect from overposting attacks, please enable the specific properties you want to bind to, for 
		//more details see http://go.microsoft.com/fwlink/?LinkId=317598.
		[HttpPost()]
		[ValidateAntiForgeryToken()]
		public ActionResult Edit(		[Bind(Include = "ID,Name")]
PlantType plantType)
		{
			if (ModelState.IsValid) {
				_db.SetModified(plantType);
				_db.SaveChanges();
				return RedirectToAction("Index");
			}
			return View(plantType);
		}

		// GET: PlantTypes/Delete/5
		public ActionResult Delete(int? id)
		{
			if ((id == null)) {
				return new HttpStatusCodeResult(HttpStatusCode.BadRequest);
			}
			PlantType plantType = _db.PlantTypes.Find(id);
			if ((plantType == null)) {
				return HttpNotFound();
			}
			return View(plantType);
		}

		// POST: PlantTypes/Delete/5
		[HttpPost()]
		[ActionName("Delete")]
		[ValidateAntiForgeryToken()]
		public ActionResult DeleteConfirmed(int id)
		{
			PlantType plantType = _db.PlantTypes.Find(id);
			_db.PlantTypes.Remove(plantType);
			_db.SaveChanges();
			return RedirectToAction("Index");
		}

		//protected override void Dispose(bool disposing)
		//{
		//	if ((disposing)) {
		//		_db.Dispose();
		//	}
		//	base.Dispose(disposing);
		//}
	}
}
